### PR TITLE
Project Awesome Books - Add the "Add book" and "Remove book" features.

### PR DESCRIPTION
**Pull request by tan12082001**
### Project: Awesome Books
Introduced features and changes:
* Collaborated as a pair programming group and created the project.
* Created an HTML page similar to the specified UI.
* Created an Array that stores the book's data (title, author) and is also used for storing and accessing the local storage.
* Added the "Add" feature which will push the new book data to an array and display the book on the webpage.
* Added the "remove" feature which will remove the book from display, the array, and local storage.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  presentbooks.innerHTML = '';
+  for (let i = 0; i < bookstorage.length; i += 1) {
+    presentbooks.innerHTML += `
+      <ul>
+          <li>${bookstorage[i].title}</li>
+          <li>${bookstorage[i].author}</li>
+          <button class='re' type='button' onclick = 'removebook(${bookstorage[i].id})'>remove</button>
+      </ul>
+    `;
+  }
```
Comment (AlphaNtihinduka) [inline app.js]: - [ ] I have noticed that on reload, the books are not retrieved from the localStorage to display on the browser. Kindly ensure that your verify, to ensure that books are displayed on reload.

File: app.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  title: this.title,
+  author: this.author,
+  id: this.id,
+};
+
+function addbook(title, author) {
+  const newbook = Object.create(bookobject);
+  newbook.title = title;
+  newbook.author = author;
+  newbook.id = Math.floor(Math.random() * 10);
```
Comment (AlphaNtihinduka) [inline app.js]: - [ ] I have noticed that you are adding the random `id` on every book created. Please ensure that the id created is unique starting from `1` to infinite. This is to ensure that the right book is deleted or edited. Also make sure that the id order is updated when a book is deleted ☺️.

**Review by AlphaNtihinduka — CHANGES_REQUESTED** [review]
## Hello Team 🙋‍♂️
Good job working on the awesome books. You have done incredibly well. 👏  

TO HIGHLIGHT:
- Books saved in the localStorage ✔️ 
- New book displayed on the page ✔️ 
- Well-structured files ✅

# STATUS: Required Changes :recycle: 

close

## Required changes :recycle: 
- _Kindly consider the inline review comments ☺️._

## Optional suggestions
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread or slack channel if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

HAPPY CODING AND GOOD LUCK :crossed_fingers: :four_leaf_clover:

File: app.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  id: this.id,
+};
+
+// Function for adding new book details
+function addbook(title, author) {
+  const newbook = Object.create(bookobject);
+  newbook.title = title;
+  newbook.author = author;
+  newbook.id = bookstorage.length + 1;
+  bookstorage.push(newbook);
```
Comment (osamaashraf6) [inline app.js:21]: - [ ] You did a great effort 😃 by implementing most of the required changes from the previous reviewer, but kindly as stated by the previous reviewer try to make sure when you deleted an item the order of ids of the items is updated as when I try to remove one item and see the order of the ids I see that it was not updated, so please try to fix that to stick to the requirements 💯🚀

> Check below to see the difference 😊⤵️
> If you can not see the image clearly just click on it to open a new tab in the browser😊

| Before deleting item| After deleting item| 
| --- | --- |
|   image |     image   |

- As you see 😃 in the image after deleting item 2 the ids are not updated in item 3 and 4 the id still as it is although the orders of id should be 1 then 2 then 3 , so please try to fix that to stick to the requirements 💯🚀

**Review by osamaashraf6 — CHANGES_REQUESTED** [review]
Hi  @tan12082001 ☺️, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

you-are-near

## Highlights ✔️

- [x] The functionality of adding items is working as expected
- [x] The items are stored in the local storage as expected
- [x] All items are displayed on the page from the local storage after adding them. 💯

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.js (hunk 3 above)
Comment (DeliceLydia) [reply app.js:21]: Addressed

**Review by DeliceLydia — APPROVED** [review]
## Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations! 🎉 

## Highlights
- Addressed requested changes



### Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**